### ISSUE TITLE
make poolmanager.conf output reproducible

### ISSUE THREAD

**Issue by calestyo** [dCache/dcache#1773]
Hi.

Every time the poolmanager.conf is save(d) it, or respectively subparts of it, seem to be arbitrarily re-ordered,... e.g. the created pools and so on.

This is kinda annoying when e.g. tracking the conf file with git, as diffs (that should only show the actual changes) are completely cluttered up.

It would be nice if the poolmanager.conf could be written in the same order (favourably version sort) for each of the respective subparts.

Also, I think one could/should drop all date/time information in the comments when the file was created.
This is already available in the mtime of the file.

Cheers,
Chris.

**Comment by calestyo** [comment]
btw: The same should be the case for commands like:
psu ls *
ls spaces
...
In earlier versions of dCache I remember that the tokens or pools were always displayed in another (random?) order... now it seems to be stable... not sure if this is just a coincidence.

I think these commands should also have some reasonable sorting, for the PoolManager commands it should probably the order of how things are set in poolmanager.conf (given that the order is pertained at save)... for the SpaceManager... don't know, based on the token ID?

An alternative would be, as mentioned above, to simply always use version sort.

Cheers,
Chris.

**Comment by gbehrmann** [comment]
https://rb.dcache.org/r/8440/

**Comment by calestyo** [comment]
Hey. Great.

Just for my information.
Is this now:
a) always keep current order (whatever that is)
b) always order, e.g. lexically
?

If it's the later, then: was it considered whether there are parts in the pool manager, where the order could actually matter (and thus (a) would be bad)?
E.g. having to links with the same prefs, that would both be usable for a read, but then dCache may take, e.g. always the first possible one?

thx,
Chris.

**Comment by gbehrmann** [comment]
If I remember correctly, if two links have the same preference, both are used (the lists of pools pointed to by the links are combined).

I chose (b) because I have been annoyed by not being able to find what I was looking for in poolmanager.conf. Keeping the existing order would solve the version control issue, but it wouldn't help with making the file easier to read. Your ticket was just the excuse I needed to actually move forward with the change. :-)
